### PR TITLE
Updated stack.yaml for all stacks

### LINKED ISSUE #62: Simplify structure of stack.yaml
Stack providers will provide meta information about the stack using `stack.yaml`.  The proposal is to keep it so minimum fields to avoid overhead today. We can always extend them back when needed.

- Proposed format:
```
name: Java MicroProfile® (title on website)
version: 0.2.2
description: Eclipse MicroProfile using AdoptOpenJDK and Maven (description on website)
license: Apache-2.0
maintainers: 
- name:
  email:
  github-id:                  
default-template: <folder-name>
```

- The stack folder name is assumed to be the `id` (when creating repo index)
- Version should be captured using `major.minor.patch` format
- Every time a stack image or template is updated the version in stack.yaml must be updated

There is no impact on updating the stack.yaml today.

### PR DESCRIPTION
**Pull request by neeraj-laad**
Updates.stack.yaml to newer format

Signed-off-by: Neeraj Laad neeraj.laad@gmail.com

### Checklist:

- [x] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [x] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [ ] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stack-structure.md).

### Modifying an existing stack:

- [ ] Updated the stack version in `stack.yaml`

Updated all stack.yaml files to match new format

### Related Issues:
Fixes #62

### CODE REVIEW

File: incubator/nodejs-express/stack.yaml [hunk 1]
```diff
@@ -1,7 +1,12 @@
-id: nodejs-express
-name: Node.js Express application
+name: Node.js Express
 version: 0.2.1
-description: Simple Node.js Express application with health checks and application metrics for Prometheus.
-maintainer: Neeraj Laad <neeraj.laad@gmail.com>
-default-template: simple
+description: Express web framework for Node.js
```
Comment (mhamwala) [inline incubator/nodejs-express/stack.yaml]: can we add Express here?

> Node.js Express

File: experimental/nodejs-functions/stack.yaml [hunk 2]
```diff
@@ -1,7 +1,9 @@
-id: nodejs-functions
-name: Node.js Functions application
+name: Node.js Functions
 version: 0.1.0
-description: Simple Node.js Functions application with health checks and application metrics for Prometheus.
-maintainer: Chris Bailey <cnbailey@gmail.com>
-default-template: simple
+description: Serverless Node.js functions
```
Comment (mhamwala) [inline experimental/nodejs-functions/stack.yaml]: can we add Express here?

> Node.js Express Functions

File: incubator/java-microprofile/stack.yaml [hunk 3]
```diff
@@ -1,7 +1,12 @@
-id: java-microprofile
-name: Java MicroProfile application built using Maven
+name: Eclipse MicroProfile®
 version: 0.2.2
-description:  Sample Java MicroProfile application built using Maven
-maintainer: Neeraj Laad <neeraj.laad@gmail.com>
-default-template: default
-license: Apache-2.0
\ No newline at end of file
+description: Eclipse MicroProfile using OpenJ9 and Maven
```
Comment (mhamwala) [inline incubator/java-microprofile/stack.yaml]: should this be:

AdoptOpenJ9

File: experimental/nodejs-functions/stack.yaml (hunk 2 above)
Comment (mhamwala) [reply experimental/nodejs-functions/stack.yaml]: or something that implies that this stack in inherited from the Node.js Express stack..

Comment (neeraj-laad) [reply experimental/nodejs-functions/stack.yaml]: I'm thinking this is what the user sees on the website tile for the stack. I don't think it matters to the user that it is based on express.. am I missing something?

**Pull request merged**